spec/PeriodicBackgroundSync-index.bs

```diff
@@ -208,54 +259,58 @@ dictionary BackgroundSyncOptions {
-
-The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">register(|tag|, |options|)</dfn></code> method, when invoked, MUST return [=a new promise=] |promise| and run the following steps:
-
-1. Let |serviceWorkerRegistration| be the {{PeriodicSyncManager}}'s associated {{PeriodicSyncManager/service worker registration}}.
-1. If |serviceWorkerRegistration|’s [=active worker=] is null, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
-1. Else, [=enqueue the following steps=] to the [=periodicsync processing queue=]:
-    1. If the {{PermissionState}} for a {{PermissionDescriptor}} with {{PermissionDescriptor/name}} <code>"periodic-background-sync"</code> is not {{PermissionState/granted}}, [=reject=] |promise| with a {{NotAllowedError}} and abort these steps.
-    1. Let |isBackground|, a boolean, be true.
-    1. For each |client| in the [=service worker clients=] for the |serviceWorkerRegistration|'s [=origin=]:
-      1. If |client|'s [=frame type=] is top-level or auxillary, set |isBackground| to false.
-    1. If |isBackground| is true, [=reject=] |promise| with an {{InvalidAccessError}} and abort these steps.
-    1. Let |currentRegistration| be the [=periodicsync registration=] in |serviceWorkerRegistration|'s [=list of periodicsync registrations=] whose [=periodicsync registration/tag=] equals |tag| if it exists, else null.
-    1. If |currentRegistration| is null:
-        1. Let |newRegistration| be a new [=periodicsync registration=].
-        1. Set |newRegistration|'s associated [=periodicsync registration/tag=] to |tag|.
-        1. Set |newRegistration|'s associated [=periodicsync registration/options=] to |options|.
-        1. Set |newRegistration|'s associated [=periodicsync registration/registration state=] to [=pending=].
-        1. Set |newRegistration|'s associated [=periodicsync registration/service worker registration=] to |serviceWorkerRegistration|.
-        1. Add |newRegistration| to |serviceWorkerRegistration|'s [=list of periodicsync registrations=].
-        1. [=Calculate a time to fire=] for |newRegistration|.
-        1. [=Schedule processing=] for |registration|.
-        1. [=Resolve=] |promise|.
+<div dfn-for="PeriodicSyncManager">
+A {{PeriodicSyncManager}} has a <dfn>service worker registration</dfn> (a [=/service worker registration=]).
+</div>
+
+  <div algorithm>
+    The <code><dfn method for=PeriodicSyncManager title="register(tag, options)">register(|tag|, |options|)</dfn></code> method, when invoked, MUST return [=a new promise=] |promise| and [=enqueue the following steps=] to the [=periodic sync processing queue=]:
+    1. Let |serviceWorkerRegistration| be the [=PeriodicSyncManager/service worker registration=] associated with the [=context object=]'s {{PeriodicSyncManager}}.
+    1. If |serviceWorkerRegistration|’s [=active worker=] is null, [=reject=] |promise| with an {{InvalidStateError}} and abort these steps.
     1. Else:
```

Comment:
Nit: Your previous step is aborting, so you don't need 'else' & nesting here.

Reply:
done.